src/components/Input/Input.jsx

```diff
@@ -0,0 +1,44 @@
+import "./Input.css";
+import { useState, useEffect } from "react";
+
+const Input = ({ id, type, children, validateForm }) => {
+  const [isValid, setIsValid] = useState(false);
+  const [isTouched, setIsTouched] = useState(false);
+
+  const changeHandler = (event) => {
+    const value = event.target.value;
+
+    if (id === "username") {
+      if (/^.+?@.+?\..+?/.test(value)) {
+        !isValid && setIsValid(true);
+      } else {
+        isValid && setIsValid(false);
+      }
+    } else if (id === "password") {
+      if (value.length >= 8 && /\d/.test(value) && /[a-zа-яё]/i.test(value)) {
+        !isValid && setIsValid(true);
+      } else {
+        isValid && setIsValid(false);
+      }
+    }
+  };
```

Comment:
The hard-coded values are no good in the reusable ui component. The ideal case would be to dynamically perform all the validation rules checked in the field code, however with no knowledge about what are these checks or how many of them are passed.

You can grab some ideas from [react-final-form field-level validation](https://codesandbox.io/s/github/final-form/react-final-form/tree/master/examples/field-level-validation) layer logics.